x-pack/plugins/security_solution/cypress/integration/detection_rules/custom_query_rule_data_view.spec.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { formatMitreAttackDescription } from '../../helpers/rules';
+import { getDataViewRule } from '../../objects/rule';
+import { ALERT_GRID_CELL, NUMBER_OF_ALERTS } from '../../screens/alerts';
+
+import {
+  CUSTOM_RULES_BTN,
+  RISK_SCORE,
+  RULE_NAME,
+  RULES_ROW,
+  RULES_TABLE,
+  RULE_SWITCH,
+  SEVERITY,
+} from '../../screens/alerts_detection_rules';
+
+import {
+  ADDITIONAL_LOOK_BACK_DETAILS,
+  ABOUT_DETAILS,
+  ABOUT_INVESTIGATION_NOTES,
+  ABOUT_RULE_DESCRIPTION,
+  CUSTOM_QUERY_DETAILS,
+  DEFINITION_DETAILS,
+  FALSE_POSITIVES_DETAILS,
+  removeExternalLinkText,
+  INDEX_PATTERNS_DETAILS,
+  INVESTIGATION_NOTES_MARKDOWN,
+  INVESTIGATION_NOTES_TOGGLE,
+  MITRE_ATTACK_DETAILS,
+  REFERENCE_URLS_DETAILS,
+  RISK_SCORE_DETAILS,
+  RULE_NAME_HEADER,
+  RULE_TYPE_DETAILS,
+  RUNS_EVERY_DETAILS,
+  SCHEDULE_DETAILS,
+  SEVERITY_DETAILS,
+  TAGS_DETAILS,
+  TIMELINE_TEMPLATE_DETAILS,
+  DATA_VIEW_DETAILS,
+} from '../../screens/rule_details';
+
+import { goToRuleDetails } from '../../tasks/alerts_detection_rules';
+import { createTimeline } from '../../tasks/api_calls/timelines';
+import { postDataView } from '../../tasks/common';
+import {
+  createAndEnableRule,
+  fillAboutRuleAndContinue,
+  fillDefineCustomRuleWithImportedQueryAndContinue,
+  fillScheduleRuleAndContinue,
+  waitForAlertsToPopulate,
+  waitForTheRuleToBeExecuted,
+} from '../../tasks/create_new_rule';
+
+import { esArchiverResetKibana } from '../../tasks/es_archiver';
+import { login, visit } from '../../tasks/login';
+import { getDetails } from '../../tasks/rule_details';
+
+import { RULE_CREATION } from '../../urls/navigation';
+
+describe('Custom query rules', () => {
+  before(() => {
+    login();
+  });
+
+  describe('Custom detection rules creation with data views', () => {
+    const expectedUrls = getDataViewRule().referenceUrls.join('');
+    const expectedFalsePositives = getDataViewRule().falsePositivesExamples.join('');
+    const expectedTags = getDataViewRule().tags.join('');
+    const expectedMitre = formatMitreAttackDescription(getDataViewRule().mitre);
+    const expectedNumberOfRules = 1;
+
+    beforeEach(() => {
+      esArchiverResetKibana();
```

Comment:
I think the spec could benefit from adding some comments from the PR description directly to the code, for instance this one could be added as a comment to this `beforeEach`:

> On the test that creates a rule using a data view, we don't call `cleanKibana` method on the before hook, instead we call `esArchiverReseKibana` on the before each one. This is because we are creating a data view we'll use after and `cleanKibana` does not delete all the data views created,  `esArchiverReseKibana` does. We don't use `esArchiverReseKibana` in all the tests because is a time-consuming method and we don't need to perform an exhaustive cleaning in all the other tests.